docs/reference-guides/interactivity-api/api-reference.md

```diff
@@ -1181,6 +1181,35 @@ This config can be retrieved on the client:
 const { showLikeButton } = getConfig();
 ```
 
+### wp_interactivity_state
+
+`wp_interactivity_state` allows the initialization of state on the server, which will be merged with with any stores defined in `view.js`.
+
+Initializing state on the server allows you to use many critical WordPress APIs, including [AJAX](https://developer.wordpress.org/plugins/javascript/ajax/), or [nonces](https://developer.wordpress.org/plugins/javascript/enqueuing/#nonce).
+
+The `wp_interactivity_state` function receives two arguments, a string with the namespace that will be used as a reference and an associative array containing the values.
+
+Here is an example of passing the WP Admin AJAX endpoint with a nonce.
+
+```php
+// render.php
+
+wp_interactivity_state(
+	'myPlugin',
+	array(
+		'ajaxUrl' => admin_url( 'admin-ajax.php' ),
+		'nonce'   => wp_create_nonce( 'myPlugin_nonce' ),
+	),
+);
+```
+
+```js
+const { state } = store( 'myPlugin', {
+	console.log( state.ajaxUrl );
+	// Expected output: https://yoursite.com/wp-admin/admin-ajax.php
+}
+```
```

Comment:
This code snippet is not correct because you're executing a console.log inside a JavaScript object.

Perhaps it would be interesting to put here a complete example (although simple) of how to use AJAX with the nonce?

```js
const { state } = store('myPlugin', {
  actions: {
    *doSomething() {
      try {
        const formData = new FormData();
        formData.append('action', 'do_something');
        formData.append('_ajax_nonce', state.nonce);

        const data = yield fetch(state.ajaxUrl, {
          method: 'POST',
          body: formData,
        }).then((response) => response.json());

        console.log('Server data!', data);
      } catch (e) {
        // Something went wrong!
      }
    },
  },
});
```

Reply:
Yes, totally appreciate the clarity here. I pushed an update to address. However, I prefer the expanded syntax (extra white-space), but I do not want to introduce a wide philsophical debate on ESLint and formatting. So, if this is not ideal formatting then let me know and I'm happy to push a different less-whitespace format. Thanks!

Reply:
Yes, in the final file, it should adhere to the WordPress coding standard. If you have your IDE set up with a Prettier extension, it should format it automatically when you format the Markdown file.

_It is not formatted with spaces here because I am using the normal Prettier when writing my GH comments, not the WordPress fork._

Reply:
Oh, I just realized that file wasn't formatted. Well, let's just merge this pull request and I'll create another one later to format it.